agent/hcp/deps.go

```diff
@@ -32,7 +32,11 @@ func NewDeps(cfg config.CloudConfig, logger hclog.Logger) (Deps, error) {
 		return Deps{}, fmt.Errorf("failed to init client: %w", err)
 	}
 
-	provider, err := scada.New(cfg, logger.Named("scada"))
+	provider, err := scada.New(logger.Named("scada"))
+	if err != nil {
+		return Deps{}, fmt.Errorf("failed to init scada: %w", err)
+	}
+	err = provider.UpdateHCPConfig(cfg)
 	if err != nil {
 		return Deps{}, fmt.Errorf("failed to init scada: %w", err)
```

Comment:
would it be helpful to have a different log message here to differentiate between this and the one above? `failed to update scada: %w`?